app/src/forms/admin/service.js

```diff
@@ -148,14 +149,35 @@ const service = {
       upd['userTokenHeader'] = null;
       upd['userTokenBearer'] = false;
     }
-
-    await ExternalAPI.query().patchAndFetchById(id, upd);
-
-    return ExternalAPI.query().findById(id);
+    let trx;
+    try {
+      trx = await ExternalAPI.startTransaction();
+      await ExternalAPI.query(trx).patchAndFetchById(id, upd);
+      await service._approveMany(id, data, trx);
+      await trx.commit();
+      return ExternalAPI.query().findById(id);
+    } catch (err) {
+      if (trx) await trx.rollback();
+      throw err;
+    }
   },
   getExternalAPIStatusCodes: async () => {
     return ExternalAPIStatusCode.query();
   },
+  _approveMany: async (id, data, trx) => {
+    // if we are setting to approved, approve all similar endpoints.
+    // same ministry, same base url...
+    if (data.code === ExternalAPIStatuses.APPROVED) {
+      const adminExternalAPI = await AdminExternalAPI.query(trx).findById(id);
+      const delimiter = '?';
+      const baseUrl = data.endpointUrl.split(delimiter)[0];
+      await ExternalAPI.query(trx)
+        .patch({ code: ExternalAPIStatuses.APPROVED })
+        .whereRaw(`"formId" in (select id from form where ministry = '${adminExternalAPI.ministry}')`)
```

Comment:
There is an SQL injection possible with this raw query. It's convoluted, but unknown if/how a bigger problem could be made:
1. Create a form "A"
2. Copy your bearer token and curl in a put request for the form data and set the ministry to `' OR 'x'='x`, so that the where clause evaluates to true no matter what the ministry is
3. Create an external API for "A"
4. Create a form "B" in a different ministry and give it the same URL for an external API
5. Approve form A's external API, and form B's external API will be automatically approved.

Hopefully the admin will notice this:
![image](https://github.com/user-attachments/assets/b9fe6311-efd8-4a46-a87c-8757ee57d645)
but perhaps there are ways to obscure it.

Reply:
That would be a bug in form API. And would be a db issue that we have no integrity on that column. So I can "block"  that here but that's not where the bugs are. Even worse, Ministry values are frontend only.  See: #1188 

Adding a bandaid to prevent sql injection in this query, but bugs should be filed and addressed.